### PR TITLE
Deploy

### LINKED ISSUE #26: Create Windows-runnable batch file to run DSpace in demo mode on local (containerized)
... with potential to upscale solution to server

### PR DESCRIPTION
**Pull request by MajoBerger**
| Phases            | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |    - |     0 |      4 |        4 |
| Developing      |  0  |   3 |    0 |      0 |         3 |
| Review             |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -   |  -    |   -    |         7 |
| ETA est.             |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -   |   -     |        0 |
## Deployment scripts
Re-used docker for our purpose.

start.*bat starts everything and creates admin account, similarly the other tasks.
Images in docker compose are taken from our company docker hub https://hub.docker.com/u/dataquest

### CODE REVIEW

File: docker/cli.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ version: "3.7"
 
 services:
   dspace-cli:
-    image: "${DOCKER_OWNER:-dspace}/dspace-cli:${DSPACE_VER:-dspace-7_x}"
+    image: "${DOCKER_OWNER:-dataquest}/dspace-cli:${DSPACE_VER:-dspace-7_x}"
```
Comment (vidiecan) [inline docker/cli.yml]: why? 
1) change to an upstream file
2) we can control it using env so why change it?

File: docker/deploy.bat [hunk 2]
```diff
@@ -0,0 +1,2 @@
+docker-compose -p our_dspace -f deploy.yml up -d
```
Comment (vidiecan) [inline docker/deploy.bat]: I would rather create "our" directory with scripts e.g., we use
`build-scripts/build` `build-scripts/test` etc.

File: docker/deploy.bat [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+docker-compose -p our_dspace -f cli.yml run --rm dspace-cli create-administrator -e test@test.edu -f admin -l user -p admin -c en
```
Comment (vidiecan) [inline docker/deploy.bat]: I use `pause` at the end of batch so I see what happend once finished

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: docker/deploy.bat (hunk 3 above)
Comment (MajoBerger) [reply docker/deploy.bat]: added

File: docker/deploy.bat (hunk 2 above)
Comment (MajoBerger) [reply docker/deploy.bat]: used build-scripts

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -24,7 +24,7 @@ services:
       DSPACE_REST_HOST: localhost
       DSPACE_REST_PORT: 8080
       DSPACE_REST_NAMESPACE: /server
-    image: dspace/dspace-angular:dspace-7_x
+    image: dataquest/dspace-angular:dspace-7_x
```
Comment (vidiecan) [inline docker/docker-compose.yml]: is this necessary?

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+on:
+  push:
+
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    runs-on: dspace-dep-1
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
```
Comment (vidiecan) [inline .github/workflows/deploy.yml]: I do not think the GITHUB_WORKSPACE is required, see https://github.com/actions/checkout

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+      - uses: actions/checkout@v3
+
+      # Ready script
+      - name: Ready script
+        run: chmod u+x $GITHUB_WORKSPACE/build-scripts/deploy-scripts/deploy.sh
```
Comment (vidiecan) [inline .github/workflows/deploy.yml]: `+x` change in git
https://stackoverflow.com/questions/21691202/how-to-create-file-execute-mode-permissions-in-git-on-windows
git update-index --chmod=+x foo.sh

File: .github/workflows/deploy.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+      - name: change, list and run
+        run: |
+          cd $GITHUB_WORKSPACE/build-scripts/deploy-scripts/
+          pwd
```
Comment (vidiecan) [inline .github/workflows/deploy.yml:20]: this is not crucial, think about using
`pwd || true`

File: .github/workflows/deploy.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: dspace-dep-1
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v3
+
+      # Ready script
+      - name: Ready script
+        run: chmod u+x $GITHUB_WORKSPACE/build-scripts/deploy-scripts/deploy.sh
+
+      - name: change, list and run
```
Comment (vidiecan) [inline .github/workflows/deploy.yml]: update name

File: .github/workflows/deploy.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+name: Deploy dspace
+
+on:
+  push:
+
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    runs-on: dspace-dep-1
```
Comment (vidiecan) [inline .github/workflows/deploy.yml:13]: add comment on what this is

**Review by vidiecan — CHANGES_REQUESTED** [review]

**Review by vidiecan — APPROVED** [review]

**Pull request merged**